packages/components/src/components/popover/popover.jsx

```diff
@@ -25,30 +30,35 @@ const Popover = ({ ...props }) => {
     }, [is_bubble_open]);
 
     const toggleOpen = React.useCallback(() => {
+        if (props.onOpen) props.onOpen();
         if (has_external_open_state) return;
         setIsPopoverOpen(Boolean(props.message));
-    }, [props.message]);
+    }, [props.message, props.onOpen]);
 
     const toggleClose = React.useCallback(() => {
+        if (props.onClose) props.onClose();
         if (has_external_open_state) return;
         if (props.is_bubble_hover_enabled) {
             setTimeout(() => {
                 // add delay to check if mouse is hovered on popover bubble
-                setIsPopoverOpen(props.is_bubble_hover_enabled ? is_bubble_open : false);
```

Comment:
We don't need to check this condition anymore?

Reply:
No. because it is inside `if (props.is_bubble_hover_enabled) {}` block.
It's always true in that line.